lib/sites/flipkart.js

```diff
@@ -0,0 +1,72 @@
+'use strict';
+
+const siteUtils = require('../site-utils');
+const logger = require('../logger')();
+
+class FlipkartSite {
+  constructor(uri) {
+    if (!FlipkartSite.isSite(uri)) {
+      throw new Error('invalid uri for Flipkart: %s', uri);
+    }
+
+    this._uri = uri;
+  }
+
+  getURIForPageData() {
+    return this._uri;
+  }
+
+  isJSON() {
+    return false;
+  }
+
+  findPriceOnPage($) {
+    // find the price on the page
+    const priceString = $('*[itemprop="price"]').attr('content');
+
+    // were we successful?
+    if (!priceString) {
+      logger.error('price not found on Flipkart page, uri: %s', this._uri);
+      return -1;
+    }
+
+    // process the price string in rupees
+    const price = siteUtils.processPrice(`R ${priceString}`);
+
+    return price;
+  }
+
+  findCategoryOnPage() {
+    // default to other
```

Comment:
I was just trying to get something up quickly to provide an example scraping implementation.  Please feel free to add to this branch to find the category.